specification/imds/data-plane/Microsoft.InstanceMetadataService/stable/2018-10-01/imds.json

```diff
@@ -0,0 +1,731 @@
+                    "properties": { 
+                        "ipAddress": { 
+                            "type":"array",
+                            "items": {
+                                "type": "object",
+                                "$ref":"#/definitions/ipv6Properties"
+                            }
+                        }
+                    }
+                },
+                "macAddress": {
+                    "type":"string",
+                    "description":"This is the MAC address of the interface."
+                }
+            }
+        },
+        "planProperties": {
+            "type": "object",
+            "properties": {
+                "name": {
+                    "type": "string",
+                    "description": "This is the Plan ID."
+                },
+                "publisher": {
+                    "type": "string",
+                    "description": "This is the publisher ID."
+                },
+                "product": {
+                    "type": "string",
+                    "description": "This is the product of the image from the Marketplace."
+                }
+            }
+        },
+        "publicKeysProperties": {
+            "type": "object",
+            "properties": {
+                "path": {
+                    "type": "string",
+                    "description": "This specifies the full path on the VM where the SSH public key is stored."
+                },
+                "keyData": {
+                    "type": "string",
+                    "description": "This is the SSH public key certificate used to authenticate with the VM."
+                }
+            }
+        },
+        "ipv4Properties": {
+            "type": "object",
+            "properties": { 
+                "privateIpAddress": {
+                    "type":"string",
+                    "description":"This is the private IP address assigned to the interface."
+                 },
+                "publicIpAddress": { 
+                    "type":"string",
+                    "description":"This is the public IP address assigned to the interface."
+                }
+            }
+        },
+        "ipv6Properties": {
+            "type": "object",
+            "properties": { 
+                "privateIpAddress": {
+                    "type":"string",
+                    "description":"This is the private IPV6 address assigned to the interface."
+                }
+            }
+        },
+        "subnetProperties": {
+            "type": "object",
+            "properties": { 
+                "address": {
+                    "type":"string",
+                    "description":"This is the address range of the subnet."
+                },
+                "prefix": { 
+                    "type":"string",
+                    "description":"This is the prefix of the subnet."
+                }
+            }
+        },
+        "instance": {
```

Comment:
Names of the model definition must be in "PascalCase". -> "Instance"

Reply:
Please add description for every model definition and for every property on the model definition. The description translates to REST API docs and SDK API docs on docs.microsoft.com website.

Reply:
Changed to PascalCase and added descriptions.